### PR TITLE
Added comments to articles

### PR DESCRIPTION
**Pull request by athifw**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -20,6 +20,14 @@ def logged_in?
   end
   helper_method :logged_in?
 
+  def require_login
+    unless logged_in?
+      flash[:error] = "You must be logged in to access this section"
+      redirect_to articles_path # halts request cycle
+    end
+  end
+  helper_method :require_login
```
Comment (yaychris) [inline app/controllers/application_controller.rb]: This isn't being used in the view, so it doesn't need to be a helper method.

File: spec/features/comments_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe "Managing comments" do
+  let!(:article) { create :article, id: 1 }
```
Comment (efatsi) [inline spec/features/comments_spec.rb]: I'm fairly confident that setting the `id` won't always take as the database has auto-incrementing ids and might ignore this setter if there's already an article in the test database. I'd suggest removing this, and referencing the article itself where you need to below:

``` ruby
visit "/articles/#{article.id}"
```

``` ruby
let!(:new_comment) { create :comment, text: 'Test Comment', article: article }
```

File: spec/features/comments_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+RSpec.describe "Managing comments" do
+  let!(:article) { create :article }
+  let!(:user) { create :user, email: 'bob@test.com', username: 'Bob' }
+  before do
+    visit '/session/new'
+    fill_in 'Email', with: 'bob@test.com'
+    fill_in 'Password', with: 'password'
+    click_button 'Log in'
+    visit "/articles/#{article.id}"
+  end
```
Comment (yaychris) [inline spec/features/comments_spec.rb:12]: This same sign in logic is duplicated across all your specs now. It's fine for now, but before your next PR, let's refactor this to remove the duplication.

**Comment by yaychris** [comment]
Looks good :turtle: :pizza:

**Comment by efatsi** [comment]
If you find some extra time it would be nice if the comments for an article appeared on the article's show page itself. Maybe you already have that on the todo list. Either way, this is nice work :+1:

**Pull request merged**